File: domain/src/main/java/com/demo/domain/base/interactors/UseCaseObserver.java
package com.demo.domain.base.interactors;

/**
 * Created by sahil on 3/17/18.
 */

import io.reactivex.observers.DisposableObserver;

public class UseCaseObserver<T> extends DisposableObserver<T> {

    @Override
    public void onNext(T t) {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onComplete() {

    }
}

File: domain/src/main/java/com/demo/domain/base/interactors/UseCaseObserver.kt
package com.demo.domain.base.interactors

import io.reactivex.observers.DisposableObserver

abstract class UseCaseObserver<T>: DisposableObserver<T>() {
    
    override fun onNext(t: T) {
        
    }

    override fun onError(e: Throwable) {

    }
    
    override fun onComplete() {

    }
    
}